File: core/core-domain/src/main/java/com/imstargg/core/domain/brawlstars/BattleEvent.java
package com.imstargg.core.domain.brawlstars;

import com.imstargg.core.domain.BrawlStarsId;
import com.imstargg.core.enums.BattleEventMode;
import com.imstargg.core.enums.BattleMode;
import jakarta.annotation.Nullable;

public record BattleEvent(
        BrawlStarsId id,
        BattleEventMode mode,
        BattleEventMap map,
        @Nullable BattleMode battleMode
) {
}

File: imstargg-frontend/src/model/BattleEvent.ts
import {BattleEventMap} from "@/model/BattleEventMap";
import {BattleEventMode} from "@/model/enums/BattleEventMode";
import {BattleMode} from "@/model/enums/BattleMode";

export interface BattleEvent {
  id: number;
  mode: BattleEventMode;
  map: BattleEventMap;
  battleMode: BattleMode | null;
}
